src/utils/emitter.ts

```diff
@@ -1,3 +1,26 @@
 import { EventEmitter } from "node:events";
+import { BotEventNames } from "../bot-events";
 
 export const defaultEmitter = new EventEmitter();
+
+/**
+ * Waits asynchronously for an EventEmitter event
+ *
+ * @param name The name of the event
+ * @param emitter The emitter that should emit the event
+ *
+ * @returns A promise that resolves to an array of event arguments
+ *
+ * @example
+ *
+ * const emitter = new EventEmitter();
+ * const promise = resolveOnEvent("my-event", emitter);
+ * emitter.emit("my-event", "content")
+ * const payload = await promise; // payload is "content"
+ */
+export const resolveOnEvent = (name: BotEventNames, emitter: EventEmitter): Promise<Array<any>> =>
+  new Promise((resolve) => {
+    emitter.on(name, (...args) => {
+      resolve(args);
+    });
+  });
```

Comment:
Good suggestion, my binary friend

Reply:
I am midway implementing this changes and it will require updating several tests unrelated with the feature. I will perform this changes in a different PR.

Reply:
Fixed in https://github.com/kleros/zodiac-bots/pull/33. Please resolve this conversation.